drivers/counter/counter_smartbond_timer.c

```diff
@@ -22,11 +25,25 @@ LOG_MODULE_REGISTER(counter_timer, CONFIG_COUNTER_LOG_LEVEL);
 
 #define TIMER_TOP_VALUE		0xFFFFFF
 
+#define COUNTER_DT_DEVICE(_idx) DEVICE_DT_GET_OR_NULL(DT_NODELABEL(timer##_idx))
+
+#define PDC_XTAL_EN (DT_NODE_HAS_STATUS(DT_NODELABEL(xtal32m), okay) ? \
+					MCU_PDC_EN_XTAL : MCU_PDC_EN_NONE)
+
+enum counter_smartbond_pm_policy_state_flag {
+	COUNTER_SMARTBOND_PM_POLICY_STATE_DRV,
+	COUNTER_SMARTBOND_PM_POLICY_STATE_PDC,
+};
+
 struct counter_smartbond_data {
 	counter_alarm_callback_t callback;
 	void *user_data;
 	uint32_t guard_period;
 	uint32_t freq;
+#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_PM_DEVICE_RUNTIME)
+	ATOMIC_DEFINE(pm_policy_state_flag, 2);
```

Comment:
do we need a bitmask here? both options seem to be mutually exclusive so a single flag that indicates whether timer is pd_tim or pd_sys would be enough.

Reply:
You are right. Fixed.